Trellix/trellix/connector.py

```diff
@@ -0,0 +1,73 @@
+"""Contains connector, configuration and module."""
+import asyncio
+import time
+from typing import Any, List
+
+import orjson
+from loguru import logger
+from pydantic import HttpUrl
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.module import Module
+
+from client.http_client import TrellixHttpClient
+
+
+class TrellixConfig(DefaultConnectorConfiguration):
+    """Configuration for TrellixConnector."""
+
+    client_id: str
+    client_secret: str
+    api_key: str
+    delay: int = 60
+    auth_url: HttpUrl = HttpUrl("https://iam.mcafee-cloud.com/iam/v1.1", scheme="https")
+    base_url: HttpUrl = HttpUrl("https://api.manage.trellix.com", scheme="https")
+
+
+class TrellixModule(Module):
+    """TrellixModule."""
+
+    configuration: TrellixConfig
+
+
+class TrellixEdrConnector(Connector):
+    """TrellixEdrConnector class to work with EDR events."""
+
+    name = "TrellixEdrConnector"
+
+    module: TrellixModule
+
+    async def get_trellix_edr_events(self) -> List[Any]:
+        """
+        Process trellix edr events.
+
+        Returns:
+            List[Any]:
+        """
+        trellix_client = TrellixHttpClient(
+            client_id=self.module.configuration.client_id,
+            client_secret=self.module.configuration.client_secret,
+            api_key=self.module.configuration.api_key,
+            auth_url=self.module.configuration.auth_url,
+            base_url=self.module.configuration.base_url,
+        )
+
+        events = await trellix_client.get_edr_investigations()
+
+        if events:
+            self.push_events_to_intakes([orjson.dumps(event).decode("utf-8") for event in events])
+
+        return events
+
+    def run(self) -> None:
+        """Runs TrellixEdr."""
+        loop = asyncio.get_event_loop()
+        try:
+            while self.running:
+                loop.run_until_complete(self.get_trellix_edr_events())
```

Comment:
The connector only fetches edr_events? That about real-time searches and historical searches?

Reply:
Realtime searches less documented ( the same as historical searches ).
So I just tested that it returns empty response, provide api calls and did not include it yet into connector processing :)

Reply:
The main problem is that it might return the same data.
Historical search we should use only on first startup, after - realtime search might be preferable. 
But still was not able to collect responses to take a look if it contains the same data in terms of events